Neos.Fusion/Classes/Core/Runtime.php

```diff
@@ -644,6 +654,9 @@ protected function getConfigurationForPath($fusionPath)
             $this->configurationOnPathRuntimeCache[$pathUntilNow]['p'] = $currentPrototypeDefinitions;
         }
 
+        // Remove any leftover internal __valueUnAssignment keys
+        unset($configuration['__valueUnAssignment']);
```

Comment:
I do not understand why this line is needed here. I really do not like that the runtime manipulating the fusion ast here. Is this really necessary.

Reply:
Technically, this is not really necessary. I mainly added this to not pollute the content cache entry identifier values. If you unset something like the GlobalCacheIdentifiers prototype (like it is done in the tests). The internal "unassignment" key is added to every cache entry identifier. 

I moved it to the content cache, so it is only removed from the entry identifier values array when rendering the identifier. Is that better?